wpk/run.sh

```diff
@@ -2,36 +2,141 @@
+    echo
+    echo "    -b,   --branch <branch>      [Required] Select Git branch or tag e.g. master"
+    echo "    -o,   --output <name>        [Required] Name to the output package."
+    echo "    -pn,  --package-name <name>  [Required for windows] Package name to pack on wpk."
+    echo "    -r,   --revision <rev>       [Optional] Revision of the package. By default: 1."
+    echo "    -p,   --path <path>          [Optional] Installation path for the package. By default: /var."
+    echo "    -j,   --jobs <number>        [Optional] Number of parallel jobs when compiling."
+    echo "    -c,   --checksum             [Optional] Whether Generate checksum or not."
+    echo "    -wk,  --wpk-key              [Optional] AWS Secrets manager Name/ARN to get WPK private key."
+    echo "    -wc,  --wpk-cert             [Optional] AWS secrets manager Name/ARN to get WPK certificate."
+    echo "    -wr,  --wpk-key-region       [Optional] AWS Region where secrets are stored."
+    echo "    -h,   --help                 Show this help."
+    echo
+    exit ${1}
+}
 
 main() {
 
+    while [ -n "${1}" ]
+    do
+        case "${1}" in
+        "-b"|"--branch")
+            if [ -n "${2}" ]; then
+                REFERENCE="$(echo ${2} | cut -d'/' -f2)"
+                shift 2
+            else
+                echo "ERROR: Missing branch."
+                help 1
+            fi
+            ;;
+        "-o"|"--output")
+            if [ -n "${2}" ]; then
+                OUT_NAME="${2}"
+                shift 2
+            else
+                echo "ERROR: Missing output name."
+                help 1
+            fi
+            ;;
+        "-r"|"--revision")
+            if [ -n "${2}" ]; then
+                REVISION="${2}"
+                shift 2
+            fi
+            ;;
+        "-p"|"--path")
+            if [ -n "${2}" ]; then
+                INSTALLATION_PATH="${2}"
+                shift 2
+            fi
+            ;;
+        "-pn"|"--package-name")
+            if [ -n "${2}" ]; then
+                PKG_NAME="${2}"
+                HAVE_PKG_NAME=true
+                shift 2
+            fi
+            ;;
+        "-j"|"--jobs")
+            if [ -n "${2}" ]; then
+                JOBS="${2}"
+                shift 2
+            fi
+            ;;
+        "-c"|"--checksum")
+            CHECKSUM="yes"
+            shift 1
+            ;;
+        "-wk"|"--wpk-key")
+            if [ -n "${2}" ]; then
+                AWS_WPK_KEY="${2}"
+                shift 2
+            fi
+            ;;
+        "-wc"|"--wpk-cert")
+            if [ -n "${2}" ]; then
+                AWS_WPK_CERT="${2}"
+                shift 2
+            fi
+            ;;
+      "-wr"|"--wpk-key-region")
+            if [ -n "${2}" ]; then
+                AWS_REGION="${2}"
+                shift 2
+            fi
+          ;;
+        "-h"|"--help")
+            help 0
+            ;;
+        *)
+            help 1
+        esac
+    done
```

Comment:
```suggestion
    while [ -n "${1}" ]
    do
        case "${1}" in
            "-b"|"--branch")
                if [ -n "${2}" ]; then
                    REFERENCE="$(echo ${2} | cut -d'/' -f2)"
                    shift 2
                else
                    echo "ERROR: Missing branch."
                    help 1
                fi
                ;;
            "-o"|"--output")
                if [ -n "${2}" ]; then
                    OUT_NAME="${2}"
                    shift 2
                else
                    echo "ERROR: Missing output name."
                    help 1
                fi
                ;;
            "-r"|"--revision")
                if [ -n "${2}" ]; then
                    REVISION="${2}"
                    shift 2
                fi
                ;;
            "-p"|"--path")
                if [ -n "${2}" ]; then
                    INSTALLATION_PATH="${2}"
                    shift 2
                fi
                ;;
            "-pn"|"--package-name")
                if [ -n "${2}" ]; then
                    PKG_NAME="${2}"
                    HAVE_PKG_NAME=true
                    shift 2
                fi
                ;;
            "-j"|"--jobs")
                if [ -n "${2}" ]; then
                    JOBS="${2}"
                    shift 2
                fi
                ;;
            "-c"|"--checksum")
                CHECKSUM="yes"
                shift 1
                ;;
            "-wk"|"--wpk-key")
                if [ -n "${2}" ]; then
                    AWS_WPK_KEY="${2}"
                    shift 2
                fi
                ;;
            "-wc"|"--wpk-cert")
                if [ -n "${2}" ]; then
                    AWS_WPK_CERT="${2}"
                    shift 2
                fi
                ;;
            "-wr"|"--wpk-key-region")
                if [ -n "${2}" ]; then
                    AWS_REGION="${2}"
                    shift 2
                fi
            ;;
            "-h"|"--help")
                help 0
                ;;
            *)
                help 1
        esac
    done
```

Reply:
Done in https://github.com/wazuh/wazuh-packages/pull/467/commits/1c82879d25119ef9aeb3808e1a4f12747d8edbf0